### PR TITLE
It is done.

### PR DESCRIPTION
**Pull request by Chawwh**
Leave a Like, Comment, and Subscribe if you enjoyed this assessment! And I'll see you next time. PEACE!

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,41 +15,105 @@
...
 
 
 // b) Create the function that makes the test pass.
+// Alrighty so I remember this question from  a couple weeks ago because it gave me nightmares! I researched ways to solve this but didn't quite understand what was going on at the time. WELLLL I do now! It was pretty nice to go back through old work and be like Ohhh! thats what that was!  
+
+// Declare a function that takes in 2 arrays
+// declare a variable for answer that is equal to [...new Set([...arr1,...arr2 ])]
+// return answer
+
+const noDupes = (arr1,arr2) => {
```
Comment (Chawwh) [inline code-challenges.test.js:109]: I TOTALLY DID THE STRETCH WRONG AAAAAA

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,41 +15,105 @@
...
+// Return shuffle
 
 
-// b) Create the function that makes the test pass.
-
 
+// b) Create the function that makes the test pass.
+const colors = (arr) => {
+    arr.shift()
+    let shuffle = arr.sort((a, b) => Math.floor(Math.random * arr.length))
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:44]: good use of sort!

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,41 +15,105 @@
...
 // Expected output: [5, 109]
 
 
 
 // b) Create the function that makes the test pass.
 
+// Declare a function that takes in an array
+// Declare a variable min and max to equal an empty array
+// push the Math.min and Math.max into minNMax using a spread operator on the array parameter
+// Return minNMax
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:71]: Awesome work! Good use of the spread operator!

File: code-challenges.test.js (hunk 1 above)
Comment (WalkerTexas-coder) [reply code-challenges.test.js:109]: lol, but you recognized it-- that counts.

File: code-challenges.test.js [hunk 4]
```diff
@@ -14,41 +15,105 @@
...
+    let answer = [...new Set([...arr1,...arr2 ])]
+    return answer
+  }
+// Yay 3 green lights!!
+
+// I was messing around to see if I was just going overboard with the spread operators but I guess I'm not? It works perfectly like this?? And I don't need multiple for loops?!? *Mind happily blown*
+
+// I hope you don't mind these shorter functions! These were actually pretty fun making and refactoring... Also, Lots of arrays this week! My goodness! I feel like a pirate writing all those Arrs!
+
+// I feel alot better about this assessment! After the last assessment I found myself feeling really defeated and doing a little rough mentally. Hopefully this week turns out better! I'm happy with my code here!
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:119]: You've done amazingly!

File: code_challenges.rb [hunk 5]
```diff
@@ -1,33 +1,93 @@
...
 num1 = 7
 # Expected output: '7 is odd'
 num2 = 42
 # Expected output: '42 is even'
 num3 = 221
 # Expected output: '221 is odd'
 
+def is_even int
+    if int.class != Integer
+        p 'Please enter a whole number'
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:23]: excellent

File: code_challenges.rb [hunk 6]
```diff
@@ -1,33 +1,93 @@
...
+end
+
+is_even num1
+is_even num2
+is_even num3
+is_even 'spaghetti'
+
+
+# I'm a little confused on how to make this method only take in integers... I'm messing around trying to get back an error when no numbers are given but it seems to be converting them somehow? Like it's trying to tell me spaghetti is even... Even if it converted the string to a number equal to the number of characters spaghetti would be odd...
+# NEVERMIND I MADE IT WORK HOW I WANTED IT TO!!! I switched my edge-case to the first if instead of the catch-all else and now it works! Hehe!
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:38]: yes! Most specific -> Least specific

File: code_challenges.rb [hunk 7]
```diff
@@ -1,33 +1,93 @@
...
 # Expected output: 'bby Rd'
 
+def no_vowels str
+    p str.delete "aeiouAEIOU"
+end
+
+no_vowels album1
+no_vowels album2
+no_vowels album3
+no_vowels 'ThE qUiCk BrOwN fOx JuMpS oVeR tHe LaZy DoG'
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:62]: look at you go

File: code_challenges.rb [hunk 8]
```diff
@@ -1,33 +1,93 @@
...
+    else 
+        puts "#{str} is not a palindrome!"
+    end
+end
+
+is_palindrome is_palindrome1
+is_palindrome is_palindrome2
+is_palindrome is_palindrome3
+is_palindrome 'Tacocat'
+is_palindrome 'Watashi wa rubi ga sukidesu'
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:93]: "ii desu ne"

File: interview-questions.md [hunk 9]
```diff
@@ -7,52 +7,57 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props are tools that allow us to transport a method or key value pair from a parent component to children components. props must be gifted through component calls.
 
-  Researched answer:
+  Researched answer: Props, short for properties, are arguments that we can pass to components. Through the use of props we can make functions and objects global!
```
Comment (WalkerTexas-coder) [inline interview-questions.md:12]: Yes!

File: interview-questions.md [hunk 10]
```diff
@@ -7,52 +7,57 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Dom events are when a user interacts with an html element that trigger a javascript execution. This can be anything from a mouse click to a reload.
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: OOP utilizes classes and methods while functional programming likes to use all functions
```
Comment (WalkerTexas-coder) [inline interview-questions.md:26]: Correct, and since we can use jsx our functions can return user relevant code using FP!

**Review by WalkerTexas-coder — APPROVED** [review]
Sahtra, hilarious and brilliant as per usual. Your testing set up is exactly what I want to be seeing and your logic is totally on point! Great catch on using the spread operator and Set(). Keep up what you are doing this is a great example of an effective work flow!

**Pull request merged**